x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -198,13 +202,72 @@ export class ManifestManager {
     );
   }
 
+  /**
+   * Builds an artifact (by policy) based on the current state of the
+   * artifacts list (Trusted Apps, Host Iso. Exceptions, Event Filters, Blocklists)
+   * (which uses the `exception-list-agnostic` SO type)
+   */
+  protected async buildArtifactsByPolicy(
+    allPolicyIds: string[],
+    supportedOSs: string[],
+    osOptions: BuildArtifactsForOsOptions
+  ) {
+    const policySpecificArtifacts: Record<string, InternalArtifactCompleteSchema[]> = {};
+    await pMap(
```

Comment:
I'm wondering if this could be done in a single `for` loop. My concern here is that 2 concurrent `buildArtifactsForOs` writes to the cache at the same time. It shouldn't happen since the cache is filled before when doing the global artifacts. But let me know what you think.

Reply:
I'm not sure I follow, but is this still a concern?